django/thunderstore/repository/models/package.py

```diff
@@ -162,8 +162,12 @@ def display_name(self):
     @cached_property
     def available_versions(self):
         # TODO: Caching
-        versions = self.versions.filter(is_active=True).values_list(
-            "pk", "version_number"
+        if hasattr(self, "unavailable_versions"):
+            del self.unavailable_versions
+        versions = (
+            self.versions.filter(is_active=True)
+            .public_list()
```

Comment:
Ideally this query should just be `self.versions.public_list()` and the implementation of `.public_list()` should account for the active status.

The planned way for the above long-term is that any changes in activity status should update the visibility flags so that queries can exclusively rely on visibility flags for filtering (and not need to account for active status or any other statuses at all).

In this case, I'd suggest overriding the `.public_list` (and potentially any other applicable visibility queryset methods) in the `PackageVersionQuerySet` class and inlining the active status filtering there, allowing the call sites to be more in line with the future plans (& thus not require changing later on).

The calling convention is a bit abnormal given the chaining nature, but something like this should work (untested, check yourself):

```python
class PackageVersionQuerySet(VisibilityQuerySet):
    def active(self) -> "QuerySet[PackageVersion]":
        return self.exclude(is_active=False)

    def public_list(self):
        return super(PackageVersionQuerySet, self.active()).public_list()
```

Reply:
In a similar vein, it might be worth overriding the `visible_list` and `visible_detail` methods while you're at it according to the planned visibility table.